code/modules/admin/verbs/randomverbs.dm

```diff
@@ -1314,6 +1314,36 @@ GLOBAL_LIST_EMPTY(custom_outfits) //Admin created outfits
 		if(ADMIN_PUNISHMENT_PIE)
 			var/obj/item/reagent_containers/food/snacks/pie/cream/nostun/creamy = new(get_turf(target))
 			creamy.splat(target)
+		if (ADMIN_PUNISHMENT_CUSTOM_PIE)
+			var/obj/item/reagent_containers/food/snacks/pie/cream/nostun/A = new(get_turf(target))
+			if(!A.reagents)
+				var/amount = input(usr, "Specify the reagent size of [A]", "Set Reagent Size", 50) as num
```

Comment:
I see all of this is quite the copypasta of https://github.com/Citadel-Station-13/Citadel-Station-13/blob/8c50b578b4e73e1b1b93e7f5359561aeac899a27/code/datums/datumvars.dm#L944

Could you please move both into a new proc for better maintanance?